velox/type/StringParser.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <fast_float/fast_float.h>
+#include <folly/Likely.h>
+#include <cmath>
+
+#include "velox/common/base/Status.h"
+
+namespace facebook::velox::util {
+static inline bool characterIsSpace(char c) {
+  return c == ' ' || c == '\t' || c == '\n' || c == '\v' || c == '\f' ||
+      c == '\r';
+}
+
+class StringParser {
+ public:
+  template <typename T>
+  static inline Status stringToFloat(const char* str, int length, T& out) {
+    if (UNLIKELY(length <= 0)) {
+      return Status::Invalid("length is less than or equal to 0!");
+    }
+    int i = 0;
+    // skip any spaces at the start
+    for (; i < length; ++i) {
+      if (!characterIsSpace(str[i])) {
+        break;
+      }
+    }
+
+    // skip any spaces at the end
+    int j = length - 1;
+    for (; j >= i; j--) {
+      if (!characterIsSpace(str[j])) {
+        break;
+      }
+    }
+
+    if (i > j) {
+      return Status::Invalid("i > j");
+    }
+
+    bool negative = false;
+    // skip leading +/-
+    switch (str[i]) {
+      case '-':
+        negative = true;
+        i++;
+        break;
+      case '+':
+        i++;
+    }
+
+    if (i > j) {
+      return Status::Invalid("i > j");
+    }
+
+    // check int/-inf and nan
+    for (int k = i; k <= j; ++k) {
+      if (str[k] == 'i' || str[k] == 'I') {
+        if (length > k + 2 && (str[k + 1] == 'n' || str[k + 1] == 'N') &&
+            (str[k + 2] == 'f' || str[k + 2] == 'F')) {
+          // Hive writes inf as Infinity, at least for text. We'll be a
```

Comment:
Why is this reference to Hive? I expect this logic not be dependent on Hive.

Reply:
You are right, I will modify the comment.